### DISCUSSION TITLE
divide-y is being applied on hidden element

### DISCUSSION THREAD

**Discussion by rockwellll** [tailwindlabs/tailwindcss#6442] [Help]
Hi!. 

So, i've a layout that looks like this

```
<ul class="divide-y-2">
  <li />
 <li />
 <li />
 <li />
 <li />

 <li class="hidden" />
</ul>
```

The divide-y utility takes in account of the hidden element and adds an extra border after it. Is there a way to not apply it on the hidden element?. Thanks

**Comment by swanson** [comment]
It is kind of hacky but you can add a `hidden` attribute 

```
<li class="hidden" hidden />
```

https://v2.tailwindcss.com/docs/upgrading-to-v2#add-hidden-to-any-template-tags-within-space-or-divide-elements

**Comment by marckohlbrugge** [comment]
Unfortunately @swanson's suggestion doesn't work in situations where you apply the `hidden` class conditionally through a variant (e.g. `md:hidden`).

There's some workarounds you can use like `[&>*:not(.md\:hidden)~*]:border-t` but the exact implementation depends on the variants you use to display the hidden tag. (in this example I'm using `md:hidden`)

Demo: https://play.tailwindcss.com/1yvT6tgzYf

Make sure to make your screen is wide enough for the `md` variant to activate.
